edge/client/interface.go

```diff
@@ -5,13 +5,20 @@ import (
 	"time"
 
 	portainer "github.com/portainer/portainer/api"
+	"github.com/portainer/portainer/api/edgetypes"
 
 	"github.com/portainer/agent"
 )
 
+type EnvironmentStatusOptions struct {
+	DoSnapshot          bool
+	DoCommand           bool
+	VersionUpdateStatus *edgetypes.VersionUpdateStatus
+}
+
 type PortainerClient interface {
 	GetEnvironmentID() (portainer.EndpointID, error)
-	GetEnvironmentStatus(flags ...string) (*PollStatusResponse, error)
```

Comment:
The point of using `flags` this way was to avoid having to couple the arguments to a fixed struct that has things that aren't useful in common (for example `DoCommand` is not useful for the regular edge poll).